lib/internal/modules/esm/get_format.js

```diff
@@ -41,15 +42,37 @@ function getDataProtocolModuleFormat(parsed) {
   return mimeToFormat(mime);
 }
 
+const DOT_CODE = 46;
+const SLASH_CODE = 47;
+
+/**
+ * Returns the file extension from a file: URL. Should give similar result than
+ * require('node:path').extname(require('node:url').fileURLToPath(url)).
+ * @param {URL} url A file: URL.
```

Comment:
If this isn't a URL instance, I think this should be

```suggestion
 * @param {URL['href']} url A file: URL.
```

Reply:
Neither work in TypeScript:

<img width="745" alt="jsdoc-url" src="https://user-images.githubusercontent.com/3012099/226169032-d3209a9b-cd22-4d13-a17c-f65faa670059.png">
<img width="744" alt="jsdoc-url-href" src="https://user-images.githubusercontent.com/3012099/226169068-bd7d5d8e-5e98-488f-bf59-07f607749e00.png">

When used within TypeScript itself (instead of a jsdoc), `URL['href']` ultimately results in the type `string`, but provides more specific info to the dev (what kind of string it should be): https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABMOcAUBDAXIgqgJQBkBtAcgAsAnAU2FIF0BKRAbwF8AoDldUgd3IYo1AG7VKpRgG4uPNBShQADgGcsAenV9tAOmoAPDAFslAG2o6IcI5JndUaACwAmaUA

```ts
function foo(a: URL['href']) {}

foo('whatever'); // ✅

foo('https://www.example.com'); // ✅

foo(42); // ❌ (TS2345): Argument of type 'number' is not assignable to parameter of type 'string'.
```